multi_agent_reinforcement_learning/envs/amod.py

```diff
@@ -1,32 +1,43 @@
 """This is the Autonomous Mobility On Demand (AMoD) environment."""
 
-from collections import defaultdict
-import numpy as np
-import subprocess
 import os
-from multi_agent_reinforcement_learning.misc.utils import mat2str
+import subprocess
+import typing as T
+from collections import defaultdict
 from copy import deepcopy
 
+import numpy as np
+from multi_agent_reinforcement_learning.data_models.actor_data import (
+    ActorData,
+    PaxStepInfo,
+)
+from multi_agent_reinforcement_learning.envs.scenario import Scenario
+
+from multi_agent_reinforcement_learning.utils.minor_utils import mat2str
+
 
 class AMoD:
     """Class for the Autonomous Mobility On Demand."""
 
     # initialization
-    def __init__(self, scenario, beta: float = 0.2):
+    def __init__(
+        self, actor_data: T.List[ActorData], scenario: Scenario, beta: float = 0.2
+    ):
         """Initialise env.
 
         scenario: The current scenario
         beta: cost of rebalancing
         """
         # updated to take scenario and beta (cost for rebalancing) as input
+        self.actor_data = actor_data
         self.scenario = deepcopy(
             scenario
         )  # I changed it to deep copy so that the scenario input is not modified by env
         self.G = (
             scenario.G
         )  # Road Graph: node - region, edge - connection of regions, node attr: 'accInit', edge attr: 'time'
-        self.demandTime = self.scenario.demandTime
-        self.rebTime = self.scenario.rebTime
+        self.demand_time = self.scenario.demand_time
```

Comment:
Something wrong here? does self.scenario lead to self.actor? If not this is a mistake right?